Update and expand GameObject Creation feature module documentation (#885)
spatialos/gdk-for-unity: docs/modules/game-object-creation/overview.md

> # GameObject Creation Feature Module

Reviewer: should this be capitalized?

Author: Its a proper noun I think

This feature module contains an default implementation of spawning GameObjects for your SpatialOS entities and offers the ability to customize that process.

## Installation

### 1. Add the package

To add this feature module to your project, add the following line to the `dependencies` section in your `workers/my-unity-project/Packages/manifest.json`:

```json
    "com.improbable.gdk.gameobjectcreation": "file:<path-to-the-gdk>/workers/unity/Packages/com.improbable.gdk.gameobjectcreation",
```

> > **Note:** `<path-to-the-gdk>` is the relative path between your `manifest.json` and the `gdk-for-unity` project.

Reviewer: this took me a while to figure out what exactly is meant here. Should we link to the gdk repo to make it absolutely clear that this is not some random project using the gdk, but the actual gdk repo.

I would also assume that many peoople are not aware that there is a `manifest.json` in the Packages folder as you would usually do it via the Unity Editor.
On that note, why not add instructions on how to add it via Unity Editor instead of the manifest.json? :D

Author: Can you add it to the manifest using the Unity Editor? That's news to me 😂

### 2. Reference the assemblies

> The GameObject Creation Feature Modules contains a single [assembly definition file](https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html) which you must reference. This process differs if you have an assembly defintion file in your own project or not.

Reviewer: ```suggestion
The GameObject Creation Feature Modules contains a single [assembly definition file](https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html) which you must reference. This process differs depending on whether you have an assembly definition file in your own project or not. 
```

> #### I have an assembly definition file

Reviewer: should we add something like 
a) I have ..
b) I don't have ...

as it looks from the preview like a new section. That might be different in the actual live docs though

Author: We can make it manually bolded section headers instead of `h4`s to make it more clearer a sub header
